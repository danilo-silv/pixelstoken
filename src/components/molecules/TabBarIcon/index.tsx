import FontAwesome from '@expo/vector-icons/FontAwesome6';
import { FunctionComponent } from 'react';
import { StyleSheet } from 'react-native';

interface TabBarIconType {
  name: React.ComponentProps<typeof FontAwesome>['name'];
  color: string;
}

export const TabBarIcon: FunctionComponent<TabBarIconType> = (props) => {
  return <FontAwesome size={28} style={styles.tabBarIcon} {...props} />;
};

export const styles = StyleSheet.create({
  tabBarIcon: {
    marginBottom: -3,
  },
});
