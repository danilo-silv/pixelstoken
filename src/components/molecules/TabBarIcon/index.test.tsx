import { render } from '@testing-library/react-native';

import { TabBarIcon } from '../TabBarIcon';

jest.mock('@expo/vector-icons/FontAwesome6', () => 'FontAwesome');

describe('TabBarIcon Component', () => {
  describe('Rendering', () => {
    it('should render FontAwesome icon correctly with given name and color', () => {
      const { getByTestId, toJSON } = render(<TabBarIcon name="home" color="blue" />);
      const icon = getByTestId('tab-bar-icon');

      expect(toJSON()).toMatchSnapshot();

      expect(icon).toBeTruthy();
    });
  });

  describe('Styles', () => {
    it('should apply correct styles to the icon', () => {
      const { toJSON } = render(<TabBarIcon name="home" color="blue" />);
      const tree = toJSON();

      expect(tree.props.style).toEqual(expect.objectContaining({ marginBottom: -3 }));
    });
  });
});
