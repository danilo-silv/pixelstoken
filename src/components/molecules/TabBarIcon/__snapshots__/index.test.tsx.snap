// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBarIcon Component Rendering should render FontAwesome icon correctly with given name and color 1`] = `
<FontAwesome
  color="blue"
  name="home"
  size={28}
  style={
    {
      "marginBottom": -3,
    }
  }
  testID="tab-bar-icon"
/>
`;
